packages/eas-cli/src/commands/update/index.ts

```diff
@@ -586,15 +613,43 @@ async function getRuntimeVersionObjectAsync(
     }
   }
 
-  return Object.fromEntries(
-    platforms.map(platform => [
-      platform,
-      nullthrows(
-        Updates.getRuntimeVersion(exp, platform),
-        `Unable to determine runtime version for ${platform}`
-      ),
-    ])
-  );
+  try {
+    return [transformRuntimeVersions(exp, platforms), null];
+  } catch (error: any) {
+    if (nonInteractive) {
+      Errors.error(error, { exit: 1 });
```

Comment:
You are still in a file where a command is defined, so it might be ok, but in general, you should avoid exiting anywhere other than top-level code and use exceptions instead.

Reply:
Yep, the common pattern for this is to just throw a normal exception. (grep for `if (nonInteractive)`).

Reply:
Sorry, it appears I was wrong (I relied on someone else comment without verifying that myself), It's fine to use `Errors.error` or `Errors.exit` anywhere in the code, it throws an exception not exiting immediately as I thought before.